enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,560 @@
+or, if a component manages something else (i.e. kubelet, kcm), adjust its behaviour (set `--cloud-provider=external` arg to kubelet for example).
+
+Specific component changes will be described in detail within separate enhancement documents on a per-component basis.
+
+#### Library Go
+
+Significant part of the code around PlatformType handling lives in the ["openshift/library-go"](https://github.com/openshift/library-go/blob/e1213f6ec5d10aa4aa8a4cac2780b1dc674c0396/pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go).
+
+Currently, this code is responsible for the decision around kubelet and kcm flags. Precisely, [IsCloudProviderExternal](https://github.com/openshift/library-go/blob/e1213f6ec5d10aa4aa8a4cac2780b1dc674c0396/pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go#L154) function
+is used for decisions around kubelet and KCM flags (within MCO and KCMO respectively).
+Also, this code is used for the decision-making about CCM operator engagement.
+
+This piece should be changed to react appropriately to the "External" platform type. During the first phases, it will need
+to behave the same as in the case of the "None" platform type. Then, in upcoming phases,
+it will need to respect additional parameters from the "External" platform spec down the road.
+
+#### Kube Controller Manager Operator
+
+Same as the [MCO](#machine-config-operator) in regard to kubelet, Kube Controller Manager Operator manages KCM (kube-controller-manager) deployments.
+
+Historically Kube Controller Manager was home for cloud-specific control loops. This logic is also engaged by setting up proper
+flags on KCM executable like,
+```shell
+...
+--cloud-provider=azure
+--cloud-config=/etc/kubernetes/cloud.conf
+...
+```
+
+For engaging an external Cloud Controller Manager no `cloud-provider` flag should be set for the KCM executable.
+
+In the context of this EP, no particular changes will be needed in the operator itself,
+changes made in [library-go](#library-go) with further dependency update should be suffice.
+
+#### Machine Config Operator
+
+Currently, MCO sets kubelet flags based on a [PlatformStatus and PlatformType](https://github.com/openshift/machine-config-operator/blob/7babc03f84088f49072c6f5b0d1769850d09a694/pkg/controller/template/render.go#L343).
+This flag is [crucial](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager) for Cloud Controller Manager engagement within the cluster.
+
+Initially, the new "External" platform should be treated similarly to PlatformType "None" by the MCO, and do not set up any cloud-specific flags for the kubelet.
+Then, down the road (during [phase 3](#implementation-phases)), it would be expected for the MCO to use the "External" platform type and its spec as a signal about the underlying platform and cloud controller manager presence and operate accordingly.
```

Comment:
WMCO will have to do the same

Reply:
Looking on current implementation, i have feeling that WMCO should just work and it depends on the MCO behaviour here: https://github.com/openshift/windows-machine-config-bootstrapper/blob/master/pkg/bootstrapper/bootstrapper.go#L382

From my understanding, some kubelet flag values (--cloud-provider [specifically](https://github.com/openshift/windows-machine-config-bootstrapper/blob/master/pkg/bootstrapper/bootstrapper.go#L487)) comes from the kubelet's systemd unit which is generated by the MCO. 

Did i missed smth, or is there any plans to change how it works?

Reply:
[Windows Machine Config Bootstrapper](https://github.com/openshift/windows-machine-config-bootstrapper/) is soon going away and will replaced by [WICD](https://github.com/openshift/windows-machine-config-operator/tree/master/cmd/daemon) but as far as the kubelet flags go it should remain the same. Just the mechanism of where it is picked up from changes (MachineConfig now vs ignition endpoint in the past). There are however other parts of the WMCO code that has some platform specific logic including for `platform:none`. cc: @sebsoto @saifshaikh48

Reply:
@aravindhp I added separate section about WMCO, PTAL: https://github.com/openshift/enhancements/pull/1234/commits/c922f9af6cfe8d6b71effb5db16c2aceeeb7eb7d